### PR TITLE
ARROW-5053: [Rust] [DataFusion] Use ARROW_TEST_DATA env var

### PR DESCRIPTION
**Pull request by andygrove**
Use `ARROW_TEST_DATA` env var instead of hard-coded relative paths.

### CODE REVIEW

**Review by sunchao — APPROVED** [review]
LGTM. Thanks @andygrove !

**Comment by sunchao** [comment]
There are failures in CI. @andygrove can you take a look?

**Review by paddyhoran — APPROVED** [review]
LGTM pending CI

**Comment by kou** [comment]
Could you set `ARROW_TEST_DATA` in `dev/release/verify-release-candidate.sh`?

**Comment by andygrove** [comment]
@kou done

File: dev/release/verify-release-candidate.sh [hunk 1]
```diff
@@ -411,7 +411,8 @@ if [ "$ARTIFACT" == "source" ]; then
   cd ${DIST_NAME}
 
   rm -r testing
-  git clone https://github.com/apache/arrow-testing.git testing
+  git clone https://github.com/apache/arrow-testing.git
+  export ARROW_TEST_DATA=$PWD/arrow-testing/data
```
Comment (kou) [inline dev/release/verify-release-candidate.sh]: If we don't use `${DIST_NAME}/testing`, `rm -r testing` is needless. And we don't put arrow-testing under `${DIST_NAME}/`.

How about this?

```diff
diff --git a/dev/release/verify-release-candidate.sh b/dev/release/verify-release-candidate.sh
index c8c631c4..4283b655 100755
--- a/dev/release/verify-release-candidate.sh
+++ b/dev/release/verify-release-candidate.sh
@@ -403,6 +403,9 @@ if [ "$ARTIFACT" == "source" ]; then
   TEST_JAVA=$((${TEST_JAVA} + ${TEST_INTEGRATION}))
   TEST_JS=$((${TEST_JS} + ${TEST_INTEGRATION}))
 
+  git clone https://github.com/apache/arrow-testing.git
+  export PARQUET_TEST_DATA=$PWD/arrow-testing/data
+
   git clone https://github.com/apache/parquet-testing.git
   export PARQUET_TEST_DATA=$PWD/parquet-testing/data
 
@@ -410,9 +413,6 @@ if [ "$ARTIFACT" == "source" ]; then
   tar xvzf ${DIST_NAME}.tar.gz
   cd ${DIST_NAME}
 
-  rm -r testing
-  git clone https://github.com/apache/arrow-testing.git testing
-
   if [ ${TEST_JAVA} -gt 0 ]; then
     test_package_java
   fi
```

**Review by kou — APPROVED** [review]
+1
I'll merge this after 0.13.0 RC4 vote is over. Because we can't verify RC4 with this change.

**Comment by kou** [comment]
Rebased.
I'll merge this after CI.

**Pull request closed without merging**